apps/dashboard/pages/identity-portal/attendee-event-scan/index.tsx

```diff
@@ -81,76 +87,62 @@ export function Index() {
       .limit(20);
 
     if (res.error) {
-      setResponse(res.error);
-      return;
+      return setResponse(res.error);
     }
 
     const logs = res.data;
-    const mappedLogs = await Promise.all(
-      logs.map(async (log) => {
-        const {
-          data: [{ first_name, last_name }],
-        } = await supabase
-          .getClient()
-          .from('user_profiles')
-          .select()
-          .eq('user_id', log.user_id);
 
+    if (logs.length > 0) {
+      // Get participant names
+      const filter = logs.map((log) => `user_id.eq.${log.user_id}`).join(',');
+      const { data: nameData, error } = await supabase
+        .getClient()
+        .from('user_profiles')
+        .select()
+        .or(filter);
+
+      if (error != null) {
+        return setResponse(error);
+      }
+
+      const newAttendees: Attendee[] = logs.map((log) => {
+        const { first_name, last_name } = nameData.find(
+          (user) => user.user_id == log.user_id
+        );
         return {
-          ...log,
-          check_in_time_display: formatTimestamp(log.check_in_time),
-          first_name,
-          last_name,
+          checkInTimeDisplay: formatTimestamp(log.check_in_time),
+          firstName: first_name,
+          lastName: last_name,
         };
-      })
-    );
-    setAttendees(mappedLogs);
+      });
+      setAttendees(newAttendees);
+    }
   }
 
-  async function updateData() {
-    const res = await supabase
+  async function updateData(userId: string, timestamp: string) {
```

Comment:
nit: add description for this function describing what it does (`updateData` seems undescriptive)

Reply:
Done!